### PR TITLE
[TD-2132] adding mdcb into auto/

### PR DESCRIPTION
**Pull request by ermirizio**
Try to add mdcb

### CODE REVIEW

File: ci/auto/dash-bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"cname": ""
+}'
+
+# 1. Create org using admin authentication
+orgid=$(curlf --header "admin-auth: 12345" \
+	      --data "$org_json" \
+	      ${db_base}/admin/organisations | jq -r '.Meta')
+
+echo "TYK_GW_SLAVEOPTIONS_RPCKEY=${orgid}"
+sed ${sed_exp} "s/TYK_GW_SLAVEOPTIONS_RPCKEY=.*/TYK_GW_SLAVEOPTIONS_RPCKEY=${orgid}/g" *.env
```
Comment (ermirizio) [inline ci/auto/dash-bootstrap.sh:50]: This variable is required for GW SLAVES to connect with MDCB. Is propagated to all env files in the root folder using SED. Not a very elegant approach

File: ci/auto/dash-bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  "active": true
+}')
+
+#2. Create user in org from (1)
+user_auth=$(curlf --header "admin-auth: 12345" \
+		  --data "$user_json" \
+		  ${db_base}/admin/users | jq -r '.Message')
+
+echo "TYK_GW_SLAVEOPTIONS_APIKEY=${user_auth}"
+sed ${sed_exp} "s/TYK_GW_SLAVEOPTIONS_APIKEY=.*/TYK_GW_SLAVEOPTIONS_APIKEY=${user_auth}/g" *.env
```
Comment (ermirizio) [inline ci/auto/dash-bootstrap.sh:67]: This variable is required for GW SLAVES to connect with MDCB. Is propagated to all env files in the root folder using SED. Not a very elegant approach

File: ci/auto/dash-bootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+sed ${sed_exp} "s/USER_API_SECRET=.*/USER_API_SECRET=${user_auth}/g" *.env
```
Comment (ermirizio) [inline ci/auto/dash-bootstrap.sh:68]: This variable plays the same role as above but, is used for MDCB tests, will get replaced into a new env file exclusively for pytest called pytest.env

File: ci/auto/local-mongo44.env [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
 TYK_PMP_UPTIMEPUMPCONFIG_COLLECTIONNAME=tyk_analytics
+TYK_PMP_ANALYTICSSTORAGECONFIG_HOST=redis
+TYK_PMP_LOGLEVEL=debug
+TYK_DB_MONGOURL=mongodb://mongo:27017/tyk_analytics
+TYK_DB_LOGLEVEL=debug
+TYK_MDCB_LOGLEVEL=debug
+TYK_MDCB_STORAGE_HOST=redis
+TYK_LOGLEVEL=debug
+TYK_GW_SLAVEOPTIONS_RPCKEY=placeholder
+TYK_GW_SLAVEOPTIONS_APIKEY=placeholder
```
Comment (ermirizio) [inline ci/auto/local-mongo44.env:14]: This variables shouldn't be here, we need to find another place that makes this sustainable.

File: ci/auto/local-postgres15.env [hunk 5]
```diff
@@ -14,3 +14,5 @@ TYK_PMP_PUMPS_SQLAGGREGATE_TYPE=sql_aggregate
 TYK_PMP_PUMPS_SQLAGGREGATE_META_TYPE=postgres
 TYK_PMP_PUMPS_SQLAGGREGATE_META_CONNECTIONSTRING="host=postgres port=5432 user=postgres dbname=tyk_analytics"
 TYK_PMP_PUMPS_SQLAGGREGATE_META_TABLESHARDING=true
+TYK_GW_SLAVEOPTIONS_RPCKEY=placeholder
+TYK_GW_SLAVEOPTIONS_APIKEY=placeholder
```
Comment (ermirizio) [inline ci/auto/local-postgres15.env:18]: This variables shouldn't be here, we need to find another place that makes this sustainable.

**Comment by ermirizio** [comment]
@alephnull I have reverted all the changes you requested and modifications have been done as well:

- Separated files compose for pro and proha
- pro-ha folder with configuration files
- Added dash-bootstrap script for mdcb config
- A taskfile has been added for developers to be able to reproduce this locally. Feel free to remove it if you consider it doesn't add any value.

**Review by alephnull — CHANGES_REQUESTED** [review]
The new env vars in local-*.env files need to find a good home.

Also, the comments on dash_bootstrap.sh need to become TODOs in the code.

**Comment by ermirizio** [comment]
In favour of https://github.com/TykTechnologies/tyk/pull/5477

**Pull request closed without merging**